### PR TITLE
Add user management

### LINKED ISSUE #10: User Management
There should be three classes of user
- Administrative Users: Has read and write access, can confer read and write access to other users.
- Standard Users: Has read and write access.
- Read only Users: Can only read data
~~- Disable and re-enable users~~

At deployment we are going to want to create a generic Intern user for them.

### LINKED ISSUE #17: Modify User Account
- [x] Users should be allowed to modify their names.

- [x] Users should not be allowed to delete their accounts to preserve record integrity.

- [x] Increase minimum password length.

### PR DESCRIPTION
**Pull request by crowesn**
Resolves #10 
Resolves #17 

Adds abilities and roles for users as well as a user admin dashboard for user management tasks. Admin role can edit name, email, and role of any user. Hides controls for actions for which users are not authorized.

* add and configure cancancan gem for abilities
* add and configure administration gem for dashboard functionality

### CODE REVIEW

**Comment by crowesn** [comment]
@hortongn @chuckgreenman 

I think this is ready to go but for the circleCI issue around the PDFkit requirement.

**Comment by hortongn** [comment]
I'll check on the circleci problem and will review.

**Comment by hortongn** [comment]
@crowesn Circleci is having trouble because the Gemfile.lock in your branch was bundled with bundler version 2.0.2, but our circleci config specifically wants to install bundler 2.0.1.

So the solution is to change this line to install 2.0.2 instead:
https://github.com/uclibs/treatment_database/blob/qa/.circleci/config.yml#L33

A more permanent solution would be to change that line to
`gem install bundler -v "$(grep -A 1 "BUNDLED WITH" Gemfile.lock | tail -n 1)"`
so that circleci will always install the version of bundler that was used to create the Gemfile.lock.

Throw that tweak into your PR and see if the build passes (a separate "Update circleci bundler version" commit is fine.  No need to squash)

File: app/controllers/users/registrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class Users
```
Comment (hortongn) [inline app/controllers/users/registrations_controller.rb]: I don't think this is how you would override #update and #destroy for users.  

Typically you would use `app/controllers/registrations_controller.rb` (no /users/) and not wrap the RegistrationsController class inside a Users class.

When I try to use the app and click the signup button I get:
```
TypeError
Users is not a class

# frozen_string_literal: true  
class Users
  class RegistrationsController < Devise::RegistrationsController     
    # before_action :configure_sign_up_params, only: [:create]     
    # before_action :configure_account_update_params, only: [:update]
```

**Comment by crowesn** [comment]
Closing see https://github.com/uclibs/treatment_database/pull/55

**Pull request closed without merging**